collector.go

```diff
@@ -0,0 +1,94 @@
+package redigosrv
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+//RedigoCollector struct to access metrics
+type RedigoCollector struct {
+	publishTrafficSize prometheus.Counter
+	methodCalls        *prometheus.CounterVec
+	subscribeAmount    prometheus.Gauge
+	subscribeSuccesses prometheus.Counter
+	subscribeFailures  prometheus.Counter
+}
+
+// RedigoCollectorOptions struct to add custom name in metrics
+type RedigoCollectorOptions struct {
+	Separator string
+	Prefix    []string
+}
+
+const (
+	//PublishMetricMethodName has the name of the specific method
+	PublishMetricMethodName string = "Publish"
+	//SubscribeMetricMethodName has the name of the specific method
+	SubscribeMetricMethodName string = "Subscribe"
+)
+
+var redigoMetricLabels = []string{"method"}
+
+//RedigoCollectorDefaultOptions will return the instance of RedigoCollectorDefaultOptions with values default
+func RedigoCollectorDefaultOptions() RedigoCollectorOptions {
+	return RedigoCollectorOptions{
+		Separator: "_",
+		Prefix:    []string{"redigo"},
+	}
+}
+
+/*
+	Output with default options:
+
+	Ex: fmt.Sprintf("%spublishTrafficSize", prefix)
+	Exit: redigo_publish_traffic_size
+*/
+
+// NewRedigoCollector will return new instance of RedigoCollector with all metrics started
+func NewRedigoCollector(opts RedigoCollectorOptions) *RedigoCollector {
+
+	prefix := strings.Join(opts.Prefix[:], opts.Separator)
+	prefix += opts.Separator
+
+	return &RedigoCollector{
+		publishTrafficSize: prometheus.NewCounter(prometheus.CounterOpts{
+			Name: fmt.Sprintf("%spublish_traffic_size", prefix),
```

Comment:
Use always `redigo` as the primary prefix.

```suggestion
			Name: fmt.Sprintf("redigo_%spublish_traffic_size", prefix),
```

Apply this to all metrics.

Reply:
Done!